hub/config/src/main/java/uk/gov/ida/hub/config/data/RemoteConfigCollectionDeserializer.java

```diff
@@ -0,0 +1,128 @@
+package uk.gov.ida.hub.config.data;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import uk.gov.ida.hub.config.domain.remoteconfig.RemoteCertificateConfig;
+import uk.gov.ida.hub.config.domain.remoteconfig.RemoteConfigCollection;
+import uk.gov.ida.hub.config.domain.remoteconfig.RemoteConnectedServiceConfig;
+import uk.gov.ida.hub.config.domain.remoteconfig.RemoteMatchingServiceConfig;
+import uk.gov.ida.hub.config.domain.remoteconfig.RemoteServiceProviderConfig;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RemoteConfigCollectionDeserializer extends StdDeserializer<RemoteConfigCollection> {
+
+    public RemoteConfigCollectionDeserializer() {
+        this(null);
+    }
+
+    public RemoteConfigCollectionDeserializer(Class<?> vc) {
+        super(vc);
```

Comment:
what does `vc` stand for?

Reply:
It stands for ValueClass and is defined by StdDeserializer from Jackson.